sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpan.java

```diff
@@ -318,6 +334,12 @@ public <T> ReadWriteSpan setAttribute(AttributeKey<T> key, T value) {
     return this;
   }
 
+  @GuardedBy("lock")
+  private boolean isSpanUnmodifiableByCurrentThread() {
+    return hasEnded == EndState.ENDED
+        || (hasEnded == EndState.ENDING && Thread.currentThread() != spanEndingThread);
+  }
```

Comment:
requires flipping the callers, but seems a bit simpler to read for me at least

(also _maybe_ ever so slightly more efficient in the typical case when updating attributes when NOT_ENDED)


```suggestion
  private boolean isModifiableByCurrentThread() {
    return hasEnded == EndState.NOT_ENDED
        || (hasEnded == EndState.ENDING && Thread.currentThread() == spanEndingThread);
  }
```